src/CSharp.Workspaces/CSharp/SyntaxLogicalInverter.cs

```diff
@@ -128,13 +128,17 @@ internal static SyntaxLogicalInverter GetInstance(Document document)
             case SyntaxKind.IsExpression:
                 {
                     var isExpression = (BinaryExpressionSyntax)expression;
-
-                    return IsPatternExpression(
-                        isExpression.Left,
-                        isExpression.OperatorToken.WithTrailingTrivia(Space),
-                        UnaryPattern(
-                            Token(SyntaxKind.NotKeyword).WithTrailingTrivia(isExpression.OperatorToken.TrailingTrivia),
-                            TypePattern((TypeSyntax)isExpression.Right)));
+                    string fullyQualifiedName = semanticModel.GetSymbolInfo(isExpression.Right, cancellationToken).Symbol!
```

Comment:
Interestingly from CSharp8 onwards, there is another case of weird behaviour that means that you cannot just replace "is" with "is not". If X is the name of the type and X is also a field on the class then `is X` points to the type and `is not X` points to the field (and hence fails to compile). See the test `Test_IsWhenIsNotIsInvalid` below for an example of this situation.

More discussion of this weirdness of the CSharp language here:
- https://github.com/dotnet/roslyn/issues/50671
- https://github.com/dotnet/csharplang/discussions/4743 

I don't know of a way to detect it without manually calculating all the in-scope meanings of the symbol in question (which I believe is non-trivial). So the best workaround that I could come up with was to specify the full qualified name to avoid any ambiguity. 

Initially, I added the `.WithSimplifyAnnotation()` so that the Simplification service would trip the fully qualified name in situations where it was necessary but unfortunately, there is a bug that means that this also won't work. I opened an issue:
https://github.com/dotnet/roslyn/issues/69426 

Aside: Is there a better way of getting a fully qualified TypeSyntax for a Symbol?

Reply:
I'd use `ToTypeSyntax().WithSimplifierAnnotation()`.

Btw, are you sure you didn't forget to add Simplifier annotation? Because I tried it locally with Simplifier annotation added and it works correctly (type name is simplified as expected).

Reply:
It must not simplify the type in the case of the test added below as that does not compile (i.e. RCS1208ReduceIfNestingTests.cs line 88).

Reply:
Please see my newer comment regarding this entire section.